### PR TITLE
feat: add BitVec.[not_not, allOnes_shiftLeft_or_shiftLeft, allOnes_shiftLeft_and_shiftLeft]

### PR DESCRIPTION
**Pull request by luisacicolini**

### CODE REVIEW

File: src/Init/Data/BitVec/Lemmas.lean [hunk 1]
```diff
@@ -1072,6 +1077,16 @@ theorem shiftLeft_add {w : Nat} (x : BitVec w) (n m : Nat) :
...
     simp at * <;> omega
 
+@[simp]
+theorem allOnes_shiftLeft_and_shiftLeft {x : BitVec w} (n : Nat) :
+    BitVec.allOnes w <<< n &&& x <<< n = x <<< n := by
+  simp [← BitVec.shiftLeft_and_distrib]
+
+@[simp]
+theorem allOnes_shiftLeft_or_shiftLeft {x : BitVec w} (n : Nat) :
+    BitVec.allOnes w <<< n ||| x <<< n = BitVec.allOnes w <<< n := by
```
Comment (tobiasgrosser) [inline src/Init/Data/BitVec/Lemmas.lean:1087]:
```suggestion
theorem allOnes_shiftLeft_or_shiftLeft {x : BitVec w} {n : Nat} :
    BitVec.allOnes w <<< n ||| x <<< n = BitVec.allOnes w <<< n := by
```

File: src/Init/Data/BitVec/Lemmas.lean [hunk 2]
```diff
@@ -1072,6 +1077,16 @@ theorem shiftLeft_add {w : Nat} (x : BitVec w) (n m : Nat) :
   cases h₅ : decide (i < n + m) <;>
     simp at * <;> omega
 
+@[simp]
+theorem allOnes_shiftLeft_and_shiftLeft {x : BitVec w} (n : Nat) :
+    BitVec.allOnes w <<< n &&& x <<< n = x <<< n := by
```
Comment (tobiasgrosser) [inline src/Init/Data/BitVec/Lemmas.lean:1082]:
```suggestion
theorem allOnes_shiftLeft_and_shiftLeft {x : BitVec w} {n : Nat} :
    BitVec.allOnes w <<< n &&& x <<< n = x <<< n := by
```

**Review by tobiasgrosser — APPROVED** [review]
LGTM

File: src/Init/Data/BitVec/Lemmas.lean [hunk 3]
```diff
@@ -939,6 +939,11 @@ theorem not_def {x : BitVec v} : ~~~x = allOnes v ^^^ x := rfl
   ext i
   simp
 
+@[simp]
+theorem not_not {b : BitVec w} : ~~~(~~~b) = b := by
```
Comment (hargoniX) [inline src/Init/Data/BitVec/Lemmas.lean:943]: Can you please remove the theorem at https://github.com/leanprover/lean4/blob/master/src/Std/Tactic/BVDecide/Normalize/BitVec.lean#L144 and tag this one with `bv_normalize` there as seen here: https://github.com/leanprover/lean4/blob/master/src/Std/Tactic/BVDecide/Normalize/BitVec.lean#L63

**Pull request merged**